backend/api/views.py

```diff
@@ -0,0 +1,56 @@
+from django.http import JsonResponse
+from rest_framework.views import APIView
+from rest_framework.permissions import IsAuthenticated
+from allauth.socialaccount.models import SocialAccount
+from allauth.socialaccount.models import SocialToken
+from api.models import APIKeyProvider
+from api.serializers import APIKeyProviderSerializer
+from rest_framework.renderers import JSONRenderer
+import requests
+
+class CredentialDetail(APIView):
+    permission_classes = (IsAuthenticated,)
+
+    def get(self, request, type):
+        if type not in ['google', 'facebook', 'octranspo', 'weather', 'openweathermap']:
+            return JsonResponse(status=400, data={'detail':'Unknown provider'})
+
+        context = {}
+
+        if type not in ['google', 'facebook']:
+            context['token'] = {}
+            data = APIKeyProviderSerializer(APIKeyProvider.objects.filter(name=type).first()).def funcname(self, parameter_list):
```

Comment:
Is this valid? It's a bit confusing. On a quick search it doesn't seem like you can bind methods like this.